scripts/update_and_rebuild_libmesh.sh

```diff
@@ -28,6 +28,21 @@ if [[ $? == 0 && "x$git_dir" == "x" ]]; then
   fi
 fi
 
+# check if the user has ccache configured
+DISABLE_TIMESTAMPS=""
+echo $CXX | cut -d ' ' -f1 | grep '^ccache$' > /dev/null
+if [ $? == 0 ]; then
+  echo -n "ccache detected - "
+  # check if timestamps are explicitly enabled
+  echo "$* " | grep -- '--enable-timestamps ' > /dev/null
+  if [ $? == 0 ]; then
+    echo "warning: setting --enable-timestamps explicitly will negatively impact the ccache performance"
+  else
+    echo "configuring limbesh with --disable-timestamps to improve cache hit rate"
+    DISABLE_TIMESTAMPS="--disable-timestamps"
+  fi
```

Comment:
Well yeah. But does it hurt to respect the user options? I can't foresee if
our why a user might want to enable timestamps for a particular build
without having to change the environment to disable ccache but I'm sure
someone will run into this. What happens if both disable and enable are
specified at the same time? Will a later enable override the disable
option? If so that logic could go.

On Fri, May 8, 2015, 08:50 John W. Peterson <notifications@github.com>
wrote:

> In scripts/update_and_rebuild_libmesh.sh
> <https://github.com/idaholab/moose/pull/5038#discussion_r29945502>:
>
> > @@ -28,6 +28,21 @@ if [[ $? == 0 && "x$git_dir" == "x" ]]; then
> >    fi
> >  fi
> >
> > +# check if the user has ccache configured
> > +DISABLE_TIMESTAMPS=""
> > +echo $CXX | cut -d ' ' -f1 | grep '^ccache$' > /dev/null
> > +if [ $? == 0 ]; then
> > +  echo -n "ccache detected - "
> > +  # check if timestamps are explicitly enabled
> > +  echo "$* " | grep -- '--enable-timestamps ' > /dev/null
> > +  if [ $? == 0 ]; then
> > +    echo "warning: setting --enable-timestamps explicitly will negatively impact the ccache performance"
> > +  else
> > +    echo "configuring limbesh with --disable-timestamps to improve cache hit rate"
> > +    DISABLE_TIMESTAMPS="--disable-timestamps"
> > +  fi
>
> This is nice of you to respect the user's options, but maybe we should
> just append --disable-timestamps to $* no matter what when ccache is
> detected... ccache is basically useless without that, right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/5038/files#r29945502>.
>